.github/workflows/CI.yml

```diff
@@ -3,9 +3,11 @@ on:
   push:
     branches:
       - "*"
+      - "feature/**"
   pull_request:
     branches:
       - "*"
+      - "feature/**"
```

Comment:
will the CI workflow will run for my forked repos too?

Reply:
In your forked repo, mostly you will be creating branch without prefix "feature/xyz" and it works by default.

Reply:
if it works by default then why we are making these changes. This is confusing.

Reply:
If my understanding is correct, the branches which we are mentioning in the workflow are the target branches to which we we are creating PRs.
Reference - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpull_requestpull_request_targetbranchesbranches-ignore

Reply:
The * wildcard matches any character, but does not match slash (/).

Reply:
> @VijayanB if it works by default then why we are making these changes. This is confusing.
> 
> `*` here should taken care of running the workflows for all the branches, but somehow that is not working.
> 
> BTW I am not a big fan of CI/CD workflows running on the forked repos.

There are two events here, push and pull request. CI runs on your forked repo because you will be creating a new branch in order to create a PR. Whereas you will see CI runs as part of PR due to pull request event on main repository.